pennylane/operation.py

```diff
@@ -964,6 +964,77 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
+    """Base class for projection operations
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all projections:
+
+    * :attr:`~.Operator.num_params`
```

Comment:
Another curiosity - do we expect `Projection` to have more than 0 params? I guess you could feed the diagonalizing unitary?

Reply:
I decided to keep it general as Nathan mentioned some possible applications of projections in QML circuits, so I thought maybe one may want to parametrize it at some point